src/tools/r2rdump/R2RDump.cs

```diff
@@ -237,37 +361,53 @@ public void DumpBytes(R2RReader r2r, int rva, uint size, bool convertToOffset =
                 _writer.Write($" {r2r.Image[start + i]:X2}");
                 if ((rva + i) % 16 == 15 && i != size - 1)
                 {
-                    _writer.WriteLine();
+                    SkipLine();
                     _writer.Write("    ");
                 }
             }
-            _writer.WriteLine();
+            SkipLine();
         }
 
-        private void DumpSectionContents(R2RReader r2r, R2RSection section)
+        private void DumpSectionContents(R2RReader r2r, R2RSection section, XmlNode contentsNode)
         {
             switch (section.Type)
             {
                 case R2RSection.SectionType.READYTORUN_SECTION_AVAILABLE_TYPES:
-                    uint availableTypesSectionOffset = (uint)r2r.GetOffset(section.RelativeVirtualAddress);
-                    NativeParser availableTypesParser = new NativeParser(r2r.Image, availableTypesSectionOffset);
-                    NativeHashtable availableTypes = new NativeHashtable(r2r.Image, availableTypesParser, (uint)(availableTypesSectionOffset + section.Size));
-                    _writer.WriteLine(availableTypes.ToString());
+                    if(!_xml)
```

Comment:
The split we have here where we have if(_xml) all over tells me we should refactor into a TextDumper / XmlDumper class. It would be nice if we could just create the right text / xml dumper class when we're parsing the input arguments and save all these ifs. In the future, if we added, say, a JSON emitter mode, it would be a matter of implementing a new set of overrides instead of having to look through all the ifs.